test/runtests.jl

```diff
@@ -79,12 +79,17 @@ using Base.Test
             @test width(ClosedInterval(A, B)) == Base.Dates.Day(59)
             @test width(ClosedInterval(B, A)) == Base.Dates.Day(0)
             @test isempty(ClosedInterval(B, A))
+            @test length(ClosedInterval(A, B)) ≡ 60
+            @test length(ClosedInterval(B, A)) ≡ 0
         end
 
         @test width(ClosedInterval(3,7)) ≡ 4
         @test width(ClosedInterval(4.0,8.0)) ≡ 4.0
 
         @test promote(1..2, 1.0..2.0) === (1.0..2.0, 1.0..2.0)
+
+        @test length(I) == 4
+        @test length(J) == 0
```

Comment:
Maybe add `@test_throws MethodError length(1.2..2.4)` just to indicate that the absence of such a method is deliberate?